cmd/istio-cni-repair/main.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A simple daemonset binary to repair pods that are crashlooping
+// after winning a race condition against istio-cni
+package main
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	client "k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"istio.io/cni/pkg/repair"
+	"istio.io/pkg/log"
+)
+
+type ControllerOptions struct {
+	DaemonPollPeriod int             `json:"daemon_poll_period"`
+	RepairOptions    *repair.Options `json:"repair_options"`
+	DeletePods       bool            `json:"delete_pods"`
+	LabelPods        bool            `json:"label_pods"`
+	CreateEvents     bool            `json:"create_events"`
+	RunAsDaemon      bool            `json:"run_as_daemon"`
+}
+
+var (
+	loggingOptions = log.DefaultOptions()
+)
+
+// Parse command line options
+func parseFlags() (filters *repair.Filters, options *ControllerOptions) {
+	// Parse command line flags
+	// Filter Options
+	pflag.String("node-name", "", "The name of the node we are managing (will manage all nodes if unset)")
+	pflag.String(
+		"sidecar-annotation",
+		"sidecar.istio.io/status",
+		"An annotation key that indicates this pod contains an istio sidecar. All pods without this annotation will be ignored."+
+			"The value of the annotation is ignored.")
+	pflag.String(
+		"init-container-name",
+		"istio-init",
+		"The name of the istio init container (will crash-loop if CNI is not configured for the pod)")
+	pflag.String(
+		"init-container-termination-message",
+		"",
```

Comment:
Why does this flag have no default value?
If you don't know what a valid default value is, no one else will.
Just remove this flag.

Reply:
This was to support the option of additionally specifying a termination message that could be written by the istio-init container by writing a message to `/dev/termination-log`. It is disabled by default (by leaving the default value as emptystring).